include/albatross/src/tune/tune.hpp

```diff
@@ -165,11 +193,19 @@ struct GenericTuner {
 
     auto grad_free_objective = [&](const std::vector<double> &x,
                                    std::vector<double> &grad) {
-      assert(grad.size() == 0 &&
-             "Gradient based algorithm used with a gradient free objective");
-
       double metric = objective(x);
 
+      if (grad.size() > 0) {
```

Comment:
This same block of code appears on ln154 above.

Reply:
it's not *exactly* the same, they both are filling in a gradient, but the actual gradient computation is different (one calls `compute_gradient` with a `ParameterStore` the other with a `std::vector<double>`.  Similarly one prints out the parameter name along side the debug output, the other only has access to the index.